src/app/gui/item/item.directive.ts

```diff
@@ -0,0 +1,50 @@
+import { ApplicationRef, ComponentFactory, ComponentFactoryResolver, ComponentRef, ElementRef, Directive, Injector, Input, Renderer2 } from "@angular/core";
+import { ItemTooltipComponent } from "../item-tooltip/item-tooltip.component";
+import { SlotComponent } from "../slot/slot.component";
+import { TooltipDirective } from "../tooltip.directive";
+import { LuJsonService, LuLocaleService } from "../../services";
+
+@Directive({
+  selector: "lux-slot[luxFetch]"
```

Comment:
I'm not quite happy with the fact that you need to pass in a LOT, but neither `lux-slot` nor `luxFetch` actually mention that. E.g. the emotes were using slots too, but they're a different kind of ID. What do you think of `[luxFetchLOT]=`?

I really appreciate the idea of separating loading from layout – but having multiple different directives with `luxFetch` feels a bit like spooky-action-at-a distance. What IDE do you use and does it succeed in sending you to this directive if you click on the `luxFetch`?

Reply:
I don't use any IDE at all, I use Notepad++ for editing :P

The idea was that you could just slap `[luxFetch]="id"` on any component that needs to be populated, no matter which one, and the matching directive would take care of the rest. For most other components this works perfectly fine, since the relevant directive would be clear from context, and it would usually just take the primary key of the relevant table, e.g. `<lux-mission [luxFetch]="id">` is pretty clear.

Slots are a bit of an edgecase since they're used for more than one thing and as such it's not entirely clear what directive would be applied, yeah. If you want I can change it to `luxFetchLOT` to disambiguate in this case.